### PR TITLE
Add erofs-util API support for overlaybd 

### PR DESCRIPTION
**Pull request by salvete**
**What this PR does / why we need it**:

Previously, erofs-utils was integrated into overlaybd by using the mkfs.erofs executable file. It's not optimial since raw data needs to be dumped first and output data needs to be write into overlaybd then.

This adds EROFS API support for overlaybd.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by hsiangkao** [comment]
relates to https://github.com/containerd/accelerated-container-image/issues/287

File: src/overlaybd/tar/erofs/liberofs.cpp [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+            LOG_ERROR("[erofs] Fail to read superblock.");
+            goto exit;
+        }
+        erofs_buffer_init(sbi, sbi->primarydevice_blocks);
+        sb_bh = NULL;
+    }
+
+    erofs_inode_manager_init();
+
+    root = erofs_mkfs_alloc_root(sbi);
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/liberofs.cpp:434]: let's rename this as erofs_rebuild_make_root(sbi);

File: src/overlaybd/tar/erofs/liberofs.cpp [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+#define SECTOR_SIZE 512ULL
+#define SECTOR_BITS 9
+#define round_down_blk(addr) ((addr) & (~(SECTOR_SIZE - 1)))
+#define round_up_blk(addr) (round_down_blk((addr) + SECTOR_SIZE - 1))
+#define min(a, b) (a) < (b) ? (a) : (b)
+#define MAP_FILE_NAME "upper.map"
+
+#define EROFS_UNIMPLEMENTED 1
+
+struct erofs_sector {
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/liberofs.cpp:20]: I think we could just use a std::map<addr, struct liberofs_inmem_sector> and stl::set for dirty bitmap?

struct liberofs_inmem_sector {
     char data[SECTOR_SIZE];
};

File: src/overlaybd/tar/erofs/liberofs.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+#include <photon/fs/filesystem.h>
+#include <photon/fs/fiemap.h>
+#include <photon/common/string_view.h>
+
+class LibErofs{
+public:
+    LibErofs(photon::fs::IFile *target);
+    ~LibErofs();
+    int extract_tar(photon::fs::IFile *source, uint64_t blksize, bool meta_only, bool first_layer);
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/liberofs.h:12]: `uint64_t blksize` should be moved to the constructor.

File: src/overlaybd/tar/erofs/liberofs.cpp (hunk 1 above)
Comment (salvete) [reply src/overlaybd/tar/erofs/liberofs.cpp]: Done.

File: src/overlaybd/tar/erofs/liberofs.cpp (hunk 2 above)
Comment (salvete) [reply src/overlaybd/tar/erofs/liberofs.cpp]: Done.

File: src/overlaybd/tar/erofs/liberofs.h (hunk 3 above)
Comment (salvete) [reply src/overlaybd/tar/erofs/liberofs.h]: Done.

**Review by hsiangkao — APPROVED** [review]
PTAL thanks @BigVan

**Comment by BigVan** [comment]
LGTM

**Comment by BigVan** [comment]
will build a new release after improving CI test

**Pull request merged**